titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/KubeNotificationProcessor.java

```diff
@@ -235,7 +235,7 @@ public void shutdown() {
                     Task fixedTask = fillInMissingStates(pod, updatedTask);
                     Task taskWithPlacementData = JobManagerUtil.attachPlacementData(fixedTask, executorDetailsOpt);
                     Task taskWithNodeMetadata = node.map(n -> attachNodeMetadata(taskWithPlacementData, n)).orElse(taskWithPlacementData);
-                    Task taskWithAnnotations = addMissingAttributes(pod, updatedTask);
+                    Task taskWithAnnotations = addMissingAttributes(pod, taskWithNodeMetadata);
```

Comment:
nice catch! Can you please add test cases to `KubeNotificationProcessorTest` so we ensure no regressions going forward?